server/aesdsocket.c

```diff
@@ -78,11 +95,262 @@ void signal_handler(int sig_num)
+ * 
+ *  @param  thread_param    thread parameters
+ * 
+ *  @return VOID
+*/
+void *client_func(void *thread_param)
+{
+    if(thread_param == NULL)
+        return NULL;
+        
+
+/* -------------------------- recv setup -------------------------- */
+    long recvLen    = 0;
+    long totalLen   = 0;
+    long currLen    = 0;
+    int res         = 0;
+
+    char recvBuf[MAX_BUF_SIZE];
+    char *recvPkt = NULL;
+    
+    struct thread_data *clientData = thread_param;
+
+    printf("\nEntered Client Thread %ld", clientData->thread_ID);
+
+/* receive packet from connected client */
+    printf("\nrecv\n");
+    while((recvLen = recv(clientData->clientFD, recvBuf, MAX_BUF_SIZE, 0)))
+    {
+        if(recvLen == -1)
+        {
+            if(errno == EINTR)
+                continue;
+            else
+            {
+                syslog(LOG_ERR, "recv failed, see errno for details");
+                printf("\ncleanup and close client thread");
+                close(clientData->clientFD);
+                clientData->clientFD = -1;
+                clientData->thread_complete_success = 1;
+            }
+        }
+
+        if(totalLen - currLen < recvLen)
+        {
+            /* add extra byte for null-term string */
+            totalLen += (recvLen + NULL_TERM_BYTE);
+            /* realloc for additional received bytes (extend) */
+            recvPkt = (char *) realloc(recvPkt, totalLen);
+            if(recvPkt == NULL)
+            {
+                syslog(LOG_ERR, "Failed to realloc memory for write buffer");
+                printf("\nrealloc for recvPkt failed\n");
+                printf("\ncleanup and close client thread");
+                close(clientData->clientFD);
+                clientData->clientFD = -1;
+                clientData->thread_complete_success = 1;
+            }
+            /* clear newly allocated extended memory */
+            printf("\nclearing realloced memory\n");
+            memset(recvPkt + currLen , 0, totalLen - currLen);
+        }
+
+        /* load recv buffer data into second buffer for file write */
+        memcpy(recvPkt + currLen, recvBuf, recvLen);
+        currLen += recvLen;
+
+        printf("\nprocessing recvPkt\n");
+        res = process_recv_pkt(&recvPkt, clientData->clientFD, clientData->mutex);
+        if(res == -1)
+            break;
+        printf("\nrecvPkt processed.\n");
+    }
+
+    /* free if packet processing failed */
+    if(recvPkt != NULL)
+    {
+        printf("\nfreeing recvPkt\n");
+        free(recvPkt);
+    }
+    
+    printf("\ncleanup and close client thread");
+    close(clientData->clientFD);
+    clientData->clientFD = -1;
+    clientData->thread_complete_success = 1;
+    return NULL;
+}
+
 int main(int argc, char *argv[])
 {
     printf("\n\nAESD Socket\n\n");
 
-    FILE *fptr;
+    struct thread_data *threadSetup = NULL;
+    struct thread_data *temp = NULL;
+    pthread_mutex_t mutex;
```

Comment:
@mash9756 missing a pthread_mutex_init for this, that's what all the valgrind errors are telling you.